app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherActivity.kt

```diff
@@ -319,32 +320,32 @@ class TabSwitcherActivity : DuckDuckGoActivity(), TabSwitcherListener, Coroutine
 
     private fun handleFabStateUpdates() {
         tabsRecycler.addOnScrollListener(
-            object : RecyclerView.OnScrollListener() {
+            object : OnScrollListener() {
                 override fun onScrolled(
                     recyclerView: RecyclerView,
                     dx: Int,
                     dy: Int,
                 ) {
                     super.onScrolled(recyclerView, dx, dy)
-                    if (dy > 20) {
-                        tabsFab.shrink()
-                    } else if (dy < -20) {
-                        tabsFab.extend()
+                    if (dy.toDp() > FAB_SCROLL_THRESHOLD) {
+                        binding.mainFab.shrink()
+                    } else if (dy.toDp() < -FAB_SCROLL_THRESHOLD) {
+                        binding.mainFab.extend()
```

Comment:
```suggestion
                    if (dy.toDp(recyclerView.context) > FAB_SCROLL_THRESHOLD) {
                        binding.mainFab.shrink()
                    } else if (dy.toDp(recyclerView.context) < -FAB_SCROLL_THRESHOLD) {
                        binding.mainFab.extend()
```

When working with window resources (as is the case here when converting to DP), always provide context if possible. This ensures that if the app is used in some multi-screen scenarios where the DP between two screens is different, that the calculation would behave as expected in context of the screen that user is currently interacting with.

Reply:
Thanks, TIL. Didn't notice those overloads before.